rust/src/lib.rs

```diff
@@ -77,13 +77,18 @@ impl Vault {
     pub async fn new(
         vault_stack: Option<&str>,
         region_opt: Option<&str>,
+        bucket: Option<&str>,
+        key: Option<&str>,
     ) -> Result<Vault, VaultError> {
         let config = aws_config::from_env()
             .region(get_region_provider(region_opt))
             .load()
             .await;
-        let cloudformation_params =
-            get_cloudformation_params(&config, vault_stack.unwrap_or("vault")).await?;
+        let cloudformation_params = if bucket.is_some() && key.is_some() {
+            CloudFormationParams::from(bucket.unwrap(), key)
+        } else {
+            get_cloudformation_params(&config, vault_stack.unwrap_or("vault")).await?
+        };
```

Comment:
Based on the Python implentation from here:
https://github.com/NitorCreations/vault/blob/60bffed8202fc12ab0ce8e0b6dc93fcc61c897da/python/n_vault/vault.py#L50-L106

Reply:
Same as comment above, from my understanding the Pytho Vault class reads the environment variables also when you are not using it through the CLI.
With this is mind, one has to then start doing the correct order of checking the existing of the env variables.